plugins/connection/lxc.py

```diff
@@ -60,20 +59,22 @@ class Connection(ConnectionBase):
     def __init__(self, play_context, new_stdin, *args, **kwargs):
         super(Connection, self).__init__(play_context, new_stdin, *args, **kwargs)
 
-        self.container_name = self._play_context.remote_addr
+        self.container_name = None
         self.container = None
 
     def _connect(self):
         """ connect to the lxc; nothing to do here """
         super(Connection, self)._connect()
 
         if not HAS_LIBLXC:
-            msg = "lxc bindings for python2 are not installed"
+            msg = "lxc python bindings are not installed"
             raise errors.AnsibleError(msg)
 
         if self.container:
             return
 
+        self.container_name = self.get_option('remote_addr')
```

Comment:
Since the container can change from task to task, the check `if self.container` above should probably also check whether `self.get_option('remote_addr')` didn't change. But that's something that shouldn't happen in this PR, since it changes the behavior of the plugin (which so far also ignored this).